### ISSUE TITLE
`defaultValue={false}` for `SelectInput` not working as expected

### ISSUE THREAD

**Issue by afilp** [marmelab/admin-on-rest#1645]
**What you were expecting:**
`defaultValue={false}` to mean that I want to literally select a `false` id in a `SelectInput`

**What happened instead:**
There is no value. If change to `{true}` there is a value.
(maybe reasonably, as `defaultValue` is a prop and `{false}` means do not show it?)

**Related code:**
https://codesandbox.io/s/r1wr199oxo
Go to `comments`, create a new comment and see the `Active` field.

```
      <SelectInput label={"Active"} source="active" choices={[
        { id: false, name: 'No' },
        { id: true, name: 'Yes' },
      ]}
        defaultValue={false}
      />
```

**Other information:**
![image](https://user-images.githubusercontent.com/7850073/37374288-153f2390-2723-11e8-8b69-a135af65a6d1.png)

**Environment**

* Admin-on-rest version:    Next, latest commit

P.S. I know I could use a `BooleanInput` but in this case the need is to show a "Yes/No" selection with text.

**Comment by fzaninotto** [comment]
I'll look into it. 

By the way, are you aware of the `NullableBooleanInput`?

**Comment by fzaninotto** [comment]
Fixed by #1655

**Comment by afilp** [comment]
About `NullableBooleanInput`, yes I was aware, but the use case requires that all created items have a value on that field (required) and that the users should not take any action if the default is the values they want. 

So your fix now makes this possible, thanks!
